src/Widgets/TimerButton.vala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2011-2018 elementary LLC. (https://github.com/elementary/camera)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ * Authored by: Alain M. <alain23@protonmail.com>
+ */
+
+public class Camera.Widgets.TimerButton : Gtk.Button {
+    private Gtk.Label timer_label;
+    private Gee.ArrayList<int> all_time;
+    private int index = 0;
+    private string disabled = _("Disable");
```

Comment:
this should probably be `private const string DISABLED` and the value should be `_("Disabled")` instead of "Disable"